packages/shared/lib/clients/orchestrator.ts

```diff
@@ -0,0 +1,406 @@
+import type { LogContext, LogContextGetter } from '@nangohq/logs';
+import { Err, Ok, stringifyError, metrics, getLogger } from '@nangohq/utils';
+import type { Result } from '@nangohq/utils';
+import { NangoError } from '../utils/error.js';
+import telemetry, { LogTypes } from '../utils/telemetry.js';
+import type { RunnerOutput } from '../models/Runner.js';
+import type { NangoConnection, Connection as NangoFullConnection } from '../models/Connection.js';
+import {
+    createActivityLog,
+    createActivityLogMessage,
+    createActivityLogMessageAndEnd,
+    updateSuccess as updateSuccessActivityLog
+} from '../services/activity/activity.service.js';
+import { SYNC_TASK_QUEUE, WEBHOOK_TASK_QUEUE } from '../constants.js';
+import { v4 as uuid } from 'uuid';
+import featureFlags from '../utils/featureflags.js';
+import errorManager, { ErrorSourceEnum } from '../utils/error.manager.js';
+import type { Config as ProviderConfig } from '../models/Provider.js';
+import type { LogLevel } from '@nangohq/types';
+import SyncClient from './sync.client.js';
+import type { Client as TemporalClient } from '@temporalio/client';
+import { LogActionEnum } from '../models/Activity.js';
+import type { TExecuteReturn, TExecuteProps } from '@nangohq/nango-orchestrator';
+
+const logger = getLogger('orchestrator');
+
+async function getTemporal(): Promise<TemporalClient> {
+    const instance = await SyncClient.getInstance();
+    if (!instance) {
+        throw new Error('Temporal client not initialized');
+    }
+    return instance.getClient() as TemporalClient;
+}
+
+export interface OrchestratorClientInterface {
+    execute(props: TExecuteProps): Promise<TExecuteReturn>;
+}
+
+export class Orchestrator {
+    private client: OrchestratorClientInterface;
+
+    public constructor(client: OrchestratorClientInterface) {
+        this.client = client;
+    }
+
+    async triggerAction<T = any>({
```

Comment:
the 2 functions `triggerAction` and `triggerWebhook` have been moved, adding a call to dryExecute. The rest of the functions remains unchanged